@xen-orchestra/lite/src/views/host/HostConsoleView.vue

```diff
@@ -0,0 +1,190 @@
+        :is-console-available="isConsoleAvailable"
+        :location="hostConsole.location"
+        class="remote-console"
+      />
+    </template>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import monitor from '@/assets/monitor.svg'
+import RemoteConsole from '@/components/RemoteConsole.vue'
+import UiSpinner from '@/components/ui/UiSpinner.vue'
+import UiStatusPanel from '@/components/ui/UiStatusPanel.vue'
+import { isVmOperationPending } from '@/libs/vm'
+import { VM_OPERATION } from '@/libs/xen-api/xen-api.enums'
+import type { XenApiHost } from '@/libs/xen-api/xen-api.types'
+import { usePageTitleStore } from '@/stores/page-title.store'
+import { useConsoleStore } from '@/stores/xen-api/console.store'
+import { useControlDomainStore } from '@/stores/xen-api/control-domain.store'
+import { useHostStore } from '@/stores/xen-api/host.store'
+import MenuItem from '@core/components/menu/MenuItem.vue'
+import MenuList from '@core/components/menu/MenuList.vue'
+import { useUiStore } from '@core/stores/ui.store'
+import {
+  faArrowUpRightFromSquare,
+  faDownLeftAndUpRightToCenter,
+  faKeyboard,
+  faUpRightAndDownLeftFromCenter,
+} from '@fortawesome/free-solid-svg-icons'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { useRoute, useRouter } from 'vue-router'
+
+const STOP_OPERATIONS = [
+  VM_OPERATION.SHUTDOWN,
+  VM_OPERATION.CLEAN_SHUTDOWN,
+  VM_OPERATION.HARD_SHUTDOWN,
+  VM_OPERATION.CLEAN_REBOOT,
+  VM_OPERATION.HARD_REBOOT,
+  VM_OPERATION.PAUSE,
+  VM_OPERATION.SUSPEND,
+]
+
+usePageTitleStore().setTitle(useI18n().t('console'))
+
+const router = useRouter()
+const route = useRoute()
+const uiStore = useUiStore()
+
+const {
+  isReady: isHostReady,
+  getByUuid: getHostByUuid,
+  hasError: hasHostError,
+  runningHosts: getRunningHosts,
+} = useHostStore().subscribe()
+const { records: getRecords } = useControlDomainStore().subscribe()
+
+const {
+  isReady: isConsoleReady,
+  getByOpaqueRef: getConsoleByOpaqueRef,
+  hasError: hasConsoleError,
+} = useConsoleStore().subscribe()
+
+const hasError = computed(() => hasHostError.value || hasConsoleError.value)
+
+const host = computed(() => getHostByUuid(route.params.uuid as XenApiHost['uuid']))
+
+const vm = computed(() => {
+  const controlDomainOpaqueRef = host.value?.control_domain
+  return controlDomainOpaqueRef ? getRecords.value.find(vm => vm.$ref === controlDomainOpaqueRef) : undefined
+})
+
+const hostConsole = computed(() => {
+  const consoleOpaqueRef = vm.value?.consoles[0]
+  return consoleOpaqueRef ? getConsoleByOpaqueRef(consoleOpaqueRef) : undefined
+})
+
+const isReady = computed(() => isHostReady.value && isConsoleReady.value && vm.value)
+
+const isHostRunning = computed(() => {
+  return getRunningHosts.value.some(runningHost => runningHost.uuid === host.value?.uuid)
+})
+
+const isConsoleAvailable = computed(() =>
+  vm.value !== undefined ? !isVmOperationPending(vm.value, STOP_OPERATIONS) : false
```

Comment:
`control domain` cannot have this kind of operation. You have to do this check on the `host` itself.
IMO, update `isVmOperationPending` to be more generic and handle both `vm` and `host` objects